include/boost/url/detail/format_args.hpp

```diff
@@ -0,0 +1,1055 @@
+//
+// Copyright (c) 2022 Alan de Freitas (alandefreitas@gmail.com)
+//
+// Distributed under the Boost Software License, Version 1.0. (See accompanying
+// file LICENSE_1_0.txt or copy at http://www.boost.org/LICENSE_1_0.txt)
+//
+// Official repository: https://github.com/boostorg/url
+//
+
+#ifndef BOOST_URL_DETAIL_FORMAT_ARGS_HPP
+#define BOOST_URL_DETAIL_FORMAT_ARGS_HPP
+
+#include <boost/url/detail/config.hpp>
+#include <boost/url/encode.hpp>
+#include <boost/url/error_types.hpp>
+#include <boost/url/url_view.hpp>
+#include <boost/url/string_view.hpp>
+#include <boost/url/detail/replacement_id_rule.hpp>
+#include <boost/url/detail/except.hpp>
+#include <boost/url/grammar/lut_chars.hpp>
+#include <boost/url/grammar/parse.hpp>
+#include <boost/url/grammar/unsigned_rule.hpp>
+#include <boost/url/grammar/variant_rule.hpp>
+#include <boost/core/ignore_unused.hpp>
+#include <boost/mp11/algorithm.hpp>
+#include <boost/mp11/function.hpp>
+#include <boost/mp11/list.hpp>
+#include <boost/mp11/utility.hpp>
+#include <boost/mp11/tuple.hpp>
+#include <array>
+
+// This file implements functions and classes to
+// type-erase format arguments.
+
+namespace boost {
+namespace urls {
+namespace detail {
+
+// state of the format string. It basically keeps
+// track of where we are in the format string.
+class format_parse_context
```

Comment:
`format_pos`?

Reply:
format_parse_context is the standard name.